python/caffe/io.py

```diff
@@ -255,7 +255,14 @@ def set_mean(self, in_, mean):
             if len(ms) != 3:
                 raise ValueError('Mean shape invalid')
             if ms != self.inputs[in_][1:]:
-                raise ValueError('Mean shape incompatible with input shape.')
+                print(self.inputs[in_])
```

Comment:
This message isn't very good debugging information.  A better message would explain that the sizes were incompatible and the mean shape was auto-resized for you, listing the sizes.